### ISSUE TITLE
Merging of FieldState's `error` and `submitError`

### ISSUE THREAD

**Issue by theneva** [final-form/final-form#95]
### Are you submitting a **bug report** or a **feature request**?

Feature request

### What is the current behavior?

`FieldState` has separate fields for `error` and `submitError` and no way to automatically merge them.

### What is the expected behavior?

`Field` should be able to handle merging of the `error` and `submitError` automatically, so the components may only handle `error`. 

From the perspective of all of my my projects', "merge" can be defined as "give `submitError` precedence". That is:

```js
function merge(error, submitError) {
 if (submitError) {
  return submitError;
 }

 return error;
}
```
I don't really see a case where you would have _both_ a valuation error (`error`) _and_ a submission error (`submitError`), so perhaps they could just be one and the same from the get-go.

However, if that's not an option (I realise that it is _possible_ to have both validation and submission errors at the same time), this could be introduced as a non-breaking change by simply allowing a prop like `mergeErrors?: boolean`, or even `mergeErrors?: (validationError: ?any, submissionError: ?any) => resolvedError?: any`—or both.

### What's your environment?

This feature does not exist in the lastest release (4.4.0 at the time of writing).

**Comment by erikras** [comment]
This doesn't seem like a concern this library should dictate. Your concern is that you find yourself doing a lot of this? (react jsx syntax)
```jsx
<div>
  {(error || submitError) && <Error>{error || submitError}</Error>}
</div>
```
and you'd rather just do
```jsx
<div>
  {mergedError && <Error>{mergedError}</Error>}
</div>
```
?

**Comment by theneva** [comment]
Thanks for your response!

To be honest, this issue was a bit… misguided. We've actually implemented `submitError` handling into our form library now instead of just theorycrafting.

We realised that it's correct (in our case) to always show the field validation `error` when present (because we default to disabling the submit button if there are any errors), and only show the `submitError` if the field validation passes.

I no longer believe it possible to "solve" this "problem" (if there even is one anymore) in this library, so I'll just close it.
